### ISSUE TITLE
[Bug]: can not filter variants in object view

### ISSUE THREAD

**Issue by Thomas-dsb** [pimcore/pimcore#12378]
### Expected behavior

Add a filter to path or name in grid view.

### Actual behavior

While trying to filter variants list on objects (path or name), en error occurs:

```
Warning: Undefined array key "type"
Server threw exception - could not perform action. Please reload the admin interface and try again
```

![filter-key-error](https://user-images.githubusercontent.com/107106611/172834077-d13d3399-000c-419d-9d6c-4dfbaceefbf6.png)

**Get Variants missing filter type**
Request URL:
https://demo.pimcore.fun/admin/object/grid-proxy?classId=CAR&folderId=79&xaction=read&_dc=1654773444689

Payload filter:
[{"operator":"like","value":"var","property":"fullpath"},{"operator":"like","property":"key"}]

**It worked on other gird views like this:**
Request URL: 
https://pimcore-test.dsbe.localhost/admin/object/grid-proxy?classId=CAR&folderId=79&xaction=read&_dc=1654772625392

Payload filter: 
[{"operator":"like","value":"var","property":"key","type":"string"}, "operator":"like","value":"d","property":"fullpath","type":"string"}]

### Steps to reproduce

Settings:
- APP_ENV=dev and APP_DEBUG=1 
- 'Allow variants' and 'Show variants in tree' is checked  

1. add some variants to an object
2. open the object with some variants
3. open tab variants
4. applay filter to path or name

**Comment by kingjia90** [comment]
The fix potentially could be similar of what was done here
https://github.com/pimcore/pimcore/pull/11939/files

**Comment by dvesh3** [comment]
Fixed by #12583
